roachtest/sqlsmith: incorrect resolution of localtimestamp overload on a remote node

roachtest.sqlsmith/setup=rand-tables/setting=no-mutations [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/6064717?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/6064717?buildTab=artifacts#/sqlsmith/setup=rand-tables/setting=no-mutations) on master @ [773f7d4445ce3e0e806b7a182adba70a0f270f19](https://github.com/cockroachdb/cockroach/commits/773f7d4445ce3e0e806b7a182adba70a0f270f19):

This logic test reproduces the problem.

```
# LogicTest: 5node

statement ok
CREATE TABLE t (c INT);

statement ok
ALTER TABLE t EXPERIMENTAL_RELOCATE VALUES (ARRAY[2], 2)

statement ok
SELECT c FROM t
WHERE
	localtimestamp(7::INT8):::TIMESTAMPTZ
	IN ('1975-04-24 08:08:35.000071+00:00':::TIMESTAMPTZ, '1980-10-15 12:17:59.000616+00:00':::TIMESTAMPTZ);
```

Jira issue: CRDB-18523

roachtest.sqlsmith/setup=rand-tables/setting=no-mutations [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/6079423?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/6079423?buildTab=artifacts#/sqlsmith/setup=rand-tables/setting=no-mutations) on master @ [0dd438d3dc0b42543890455945a7a6b42811def1](https://github.com/cockroachdb/cockroach/commits/0dd438d3dc0b42543890455945a7a6b42811def1):


```
test artifacts and logs in: /artifacts/sqlsmith/setup=rand-tables/setting=no-mutations/run_1
	sqlsmith.go:265,sqlsmith.go:325,test_runner.go:896: error: pq: internal error: unable to vectorize execution plan: overlaps(localtimestamp("$1" + -1522413462603336725), '2004-05-17 12:24:11.000547+00:00':::TIMESTAMPTZ, '1970-01-20 01:53:09.000019+00:00':::TIMESTAMPTZ, '2029-05-09 02:21:44.000306+00:00':::TIMESTAMPTZ): unknown signature: overlaps(timestamp, timestamptz, timestamptz, timestamptz)
		stmt:
		WITH
			with_32621 (col_180414)
				AS (
					SELECT
						*
					FROM
						(VALUES ('rrhtr':::rand_typ_3), ('rrhtr':::rand_typ_3), ('rrhtr':::rand_typ_3), ('rrhtr':::rand_typ_3))
							AS tab_79205 (col_180414)
				),
			with_32622 (col_180415)
				AS (
					SELECT
						*
					FROM
						(
							VALUES
								(NULL),
								(
									'01030000C0010000000B0000002484884266FBF7C13E6AA8E91639E2C1502244AF12C4E3410DFE2CE86A6AF2C108BE6B6F49A5D2C11F6FAFB925D2F9C18094D59CA532E1C1A8662D3C4EB3F04110613D75B8D3CCC12C3FC16DC00B00C2801AFEFA372DBF411027DC592F37E14104B3FB7E977CE541DE8F1103A04EF6C1008EB39789B3CCC1B819DCA73C24DDC1A09438FFE4E001424B996B7AEF6500C2489C1FDF7D6EEEC12FAE61CE59CDFDC170117EC4BB29FD41909B991473EFC9419870F1E4A4AAD34106CEA19F4645F3C1F0F43AB5648CCEC1762356DAADC2F1418C0A68AD8421ED41D469BD156329F1412E4510149D18FCC1AA9FBBBAD230F041DC0CF9B176D0FEC1A89D9A5969A9F7C1BF2457DC4170F2C190E245C429E1D441B6ACD3724F37F34132C8161A0EFFF0C1877A30669035FEC10060E15BA9F87341D0E8F66928B5D2C15090E58989A301422484884266FBF7C13E6AA8E91639E2C1502244AF12C4E3410DFE2CE86A6AF2C1':::GEOMETRY
								),
								(
									'0103000040010000000B00000082CAFB2488C3FEC15A62C5904C1B02C218B6768B1AE3EC4150CAFE47F736CB41A0A196A6E95A02C200B8CB49DE5D02C210E7510C13DCFC415BEADE1C378AFEC14078A91B263FD34120D53214F6F4F6414670A4A5D564F3C1DE1FCD1CF3AEFCC1545807CF069EEF4120BEB0BD2E99C1C128ADF18C396DD3C1E0F194CCC395F241C84A23CF0B38D341E0C656AB595DC6415625DD1AF34BF841E23812A13512F44124DE72CE8D0AECC1F02BB3376B1DC0C132B39FDE0F56F441182FDF5A9A3602424DF5624A80ECF7C14C872ACA04EAF641B4F27CB31B46E041A87CE4CB0A41F2C1F0458707BDD8CD41D0154CDB0461FF4182CAFB2488C3FEC15A62C5904C1B02C218B6768B1AE3EC41':::GEOMETRY
								),
								('01050000C000000000':::GEOMETRY),
								(
									'01030000C001000000080000004C29A741934DF7C188BE81090DD1EBC11CCFEFCF03B7E24180CFB7855D93E041500BCB8B4553D441CBE5797204D601C2D54419F9F91F02C2467DC2BC7DA8F141E0E01DC2270CFA41347CCCC3EBFDEDC1A8A7AEE1DBACD3C132DAB252B9DA01C2BC82709DC22FFE41D60F9BA8B6DFF341349E64E00818F841ECEA9958533CEF417092C3AB78F8C64134C54178EB98F341706B0A981C4CF7415CC1E0289E73E84100D4FC859675A2C1CA311F69DB32F1418071CB2981A8F8415CF5FA4F8203E241AE7907846D7EEBC1C8BA8514EE3E014200204CEF9542C5C160A17BED21BAC7C14C29A741934DF7C188BE81090DD1EBC11CCFEFCF03B7E24180CFB7855D93E041':::GEOMETRY
								),
								(
									'0107000080090000000106000080010000000103000080010000000A00000092638831D98AFAC1D00BB61E42A6CFC1C0AF61B86ADAEDC1C0589B4B34BBB94106F594F3269CE8C19A1552748D3202423C890E28F4B7F0413EBDE4293C21EFC1ACFB8DF933B50142283A25CDA139F4417C125411E65CEEC1B83FD4784063F941FACACE3D0415F94150E545B57CEBEF41F4BB7825105EFF4174505959C3ECF141349635E04782ED415A2A754CDA30F841A8BC484CD4A8F941406D5C55B5FC00429A781D2A91D100423277D4F91745F0C162C978BECB2102427432D86F42A8E641E001273252C7BD41CC0C50B139D1E041609EA93D7DB001C292638831D98AFAC1D00BB61E42A6CFC1C0AF61B86ADAEDC10102000080020000005048E61F2FF6E54178156F02CDBFDF41D32294A87513FBC14477A07EF8A0F3C148AC93554C35E441D2CA0A2A8C9D0142010500008009000000010200008005000000FA6AC10028B6F8C1B01EB29D17DED8C1810B6EDDB98702C23CBE3D4AA111EAC10C201C7E4F7DDEC18085031359F89D4128D319E2373CDBC1CB20774EC663F3C152CE1E0B9ED4F841509C7E254B7FD2C130FB0F53FB57EEC1301D83180D13004210F9993FBAE1CF411A7001CF7E0EEAC152592F387D6CF7C1010200008003000000B885F2684259D44128BE079A5CBBD2C18456090B0BEBF9C120179C3A825DF34171A68077472AFBC1D42F7E6CDFEFEAC1FC378B552D93F141DE0863A437CEF341FBA29B37AD33F2C10102000080020000005DDE0E53705F00C2100D17AEE00ACA410ADDD89C806DFAC104BF3926DF04F4C100EA5A295576C2412862092A5B67F6C1010200008004000000FAA202B7886EECC1796DACF78FE0FEC1A8D1CA1007A5FB41149C222C2B69E541D12993C5A0C4F3C1540F4F41A2A1F5C1C8497B66368AD0415656E04594780142404FD8D373D6D0C1AE4C73029227EEC1BEDA8476DB57F04100EEE506CF6B73C101020000800200000094A53F2BCAE2EC4186BE38033B1501C28C8C254897E8FA41E8304161DC000042E45601E5152DEA4124987DD3338DF2410102000080030000000C0AFE1035C9FC41FD12F9A3E29002C2808F5DE123F99F4112739940024DFA412EEF8F4A79C4E9C18572010E21DFF5C19C094BCAADB3F8C1401189076017D2C1E084BC0C69AAD3C10102000080020000005CFF1195A83FF141D81FB40472BEE0416CC8F1AA799AF5416857F4220DF0EF41ECBA8F05A59FEE41E810505A4588EEC1010200008003000000E2E1A71343B9004200CC25B27C49AB4103D3E52525E8F2C13C2F7ADE3ECCF841B0C0D65C73D4E4410211E8B343DF01422268E89E93DAE0C100BA6BA7565BE941FC9FC2FDD37BED410102000080040000002C8AC6D31486E041ACF37811E43CF6C10C6B92CBD1F4FB4188ED64F83A82F5416CFBFF836100F7C1AC92D568FDDBFDC14016095C20E3FB413AC8B3ABD20BFFC198C760BB2BFFF7C1184E94B5522F0042143675CE618100C2AEC61EB1AD2601420103000080010000000900000066BA180939C0EFC19BAB2B78AABDF2C194A6866F6DC30142AACFE5D2F006E7C18411D0874B0BEAC1E025FAAEB411E1C1F014A88B936BC841E8B947C416F2FFC1F6517C8AD7F9F64100BC234A23C3DB4148949EC3F4EFD6C1906043AD59B6D3C160EDD4CD15A4C341C0E0FE533E75BA412896D38D2AD9014262CE32EDB03EE7C1A2C1EAFC6ABAF5412C26FD01ECD7DFC132EABFDAC6ADFFC1C0DAAEC5D3FAB241DA1BAB24317702428165844EBF83FAC10051D2A5C95FCAC174B8012BA8E9FE4166BA180939C0EFC19BAB2B78AABDF2C194A6866F6DC301420106000080090000000103000080010000000B000000F741DCB491A100C2F04DE2A95C4ADD41B44895EE4D2FDDC1FC715357E7B6F5C180DB6ACDF837C641F4C77DFB7437E5C18046443F16B7B8C1234EB070E7F8F7C1AA66AA0D53F2EBC160D08C7C269EE64100CAE806341CC041909135571E61F541789F33C8E148F941E0D38CCB5F03DF4118E8467307DDF541A662225898D4F241B8A64E22FF42EC41D0CC1FAE5C16DAC14CB4B039DD22DEC146192F27A7F3F741C853A02F117AFA419AA1A7B37218E5C1369F547D8C74FA410A47E47FF4B3F3C1B13689EE0030F2C1DCD0BF390AD9F341F30E678FEA32FFC104C04C2C64D4FCC130F0A4B14546E2415860D9E8A62DF541F741DCB491A100C2F04DE2A95C4ADD41B44895EE4D2FDDC101030000800100000008000000D2E5C64221E8F7C16807BA12C930FCC1E6184F01CA3A004212B2842EBD81EFC1CC6B1D365A7DF6C1844EBBB92C3BFA411AAB30B8C7C0F54170D7392C4BBEC2411246BC82D58F00C260333FA08751F541807380C619FAE6419C389E3FE1CFF6412A17312CA0D6F6416A983001DF6F0242A07AC7A35317F441006926D0A364B441788C8F25471EF741C82404AFF3DA00421449F79F030ED6C1C8E7D56D770DFD41CC67D7DA2C690042D2E5C64221E8F7C16807BA12C930FCC1E6184F01CA3A004201030000800100000004000000E41E98134B05E94161292C445FE9FAC13EF0D6BE9F04F841B8E49A62B8BF0142003FF3C55578B5C1D09845C9A943DEC1E223F6330A83F1C1027CCC406171F541FCCCD4532DB9F8C1E41E98134B05E94161292C445FE9FAC13EF0D6BE9F04F84101030000800100000007000000907C62D54BDFE0C160FC29DA7137D9416C2B0063A3B9E9410817B8012E53D8C13754D8CF9E3802C26CAE5ABF13E2F7C18C5EB6A96D5BFC4150CBB9DE691BC14168698FCD65C5F041CA79254C9D020142820EE9F569810042708FC9AD9106F541FA836FB45F0EF44102DE6612DF06F7419CE91310F947F541404210B4BC0ACB41A2DFB5F7D20EF7412290E1B3C222F7C1907C62D54BDFE0C160FC29DA7137D9416C2B0063A3B9E94101030000800100000004000000EB4F3F3C7A89F9C1FF6329293434FFC1D4BC0C2F8021D9C1B879E50F4E7DEC41406DBF5EDB4AFC41A0EBF0DF3223BA410C708C624ECDDDC198F5C3D5BFADFD41AD5E52C203FEFCC1EB4F3F3C7A89F9C1FF6329293434FFC1D4BC0C2F8021D9C10103000080010000000A0000008A9F60337476FBC15084039F5847F8C17836CDE7346CFFC12D04F211A7EAF1C1C6C59757C18AF6C15C3E373AFA38F641F21C063B2C24F5C14916D4C5121EFEC150D560868179DD41E8D8A1BE4AAEDCC1323CACA0EFE4F5C1743D6B56174CF6419A9F1AA75A6CF14136894A7CD621F3C1EA031072DC9CE0C15CDB1F93763902428435DF835B74EC415A9764216B8DF5410C58F0108862F241D81FA47A5618D5419487AB985CC7F2C18853D28273E4EB41F072880F0FDDCF416241FB960B3E02429C4FF366308EF14104C878F1CEFC00422041615500F2BE418A9F60337476FBC15084039F5847F8C17836CDE7346CFFC101030000800100000004000000ACAF12D6C798E441C02A1B7173AECF413AA7B1E469E9F541C83D8640BC67D0C138D6F3894C92E94138733C1F6000DDC119F1FD6014CA00C224CF76D26304E84144EEDF4F487DF041ACAF12D6C798E441C02A1B7173AECF413AA7B1E469E9F541010300008001000000050000000EC30A729E15F1C1E85D666EA2E6E3C1226308673676FA41907C6AC1BC1EC54140D9AAFAA1FCCFC1D305F91EE0C8F1C1C008EBF46254D441D4D277814181E041EDF7B42CCF7AF3C1D5345EFB5F1E02C2E06AF55C29CAB6C16F50C783761BF1C10EC30A729E15F1C1E85D666EA2E6E3C1226308673676FA41010300008001000000090000007E193D288037FCC10CF1CB572734F6C1C07061252C17BD4146E35BDB221502C22E247A27624701C2708FD2C43675C7C17512FE36C961FEC1AB2F9E5CE11100C2DCC16F46B998F441F672078EB58CF641A566A71CB44101C202BDEE74C7CFF8C180DDC1571EC597C1C817EE92B165EE41A4A8E3BE702CF741300F2170EEDAC741B0C7CDE6AA14FE41041C49173A63FB4160A350890644BEC1606818C34F4D004220D269ECE324E841DC632FF51104FFC17A3A316A02F2F5416CF52EC41D67FB417E193D288037FCC10CF1CB572734F6C1C07061252C17BD4101030000800100000007000000FA91DC9B8346EAC1049845A45FC1E1C10450578C87AAFB41F893FB494A63F5C1B858BBD8CED7F9C16C1211746D5EFB41FC4B1E0E9294D7C1AAFF0C1AC9B7FAC1744FBFEAFBABF5415081C8F31558F94120DFD2F0475ABFC140C16E29C4F7A3C19086929C99A9ED41AC1CDE3D21BF00420C84848B33EBFC4180B1FA6D0B78FBC150815557AF4B0042046D842F0A38E9C1FA91DC9B8346EAC1049845A45FC1E1C10450578C87AAFB410105000080070000000102000080030000001F6CEA94E95001C2619863C7E64EFDC132C6095EC87FE8C1ECA9CDCFA80DFB41D42424BC824ADAC17FA576A4668900C268E949422DA8EF415C759DBC95B0EB41909B4A670E06EE4101020000800400000030CD60B6107ED24152E819B6DD65FBC1A6F7F188A552FA418CC52997F3A3FA413EAE7E495D8700C2205380080BD4E6C1800E34687DD6FB41D9BB12E03ADDF0C116345DF890200142E655DFA1A77CF8411646989F54B2E8C180097E70160CE3C1010200008002000000F2B89CCA40E7014288EB0D45A75FDD4150288DF848CBED4130898EC052C2FA416C47F524FFB3F541008CB6F14C9C9341010200008003000000FA1E7A1E2968F8415F28750FCA6102C29F9C84B0A51C02C27872D80C70290042F8D8CACFB8A7DB412C67EDBAFBB7014200443F45C42A8EC1B89E53865102EB41AEF79257A37C0142010200008004000000B4C889C6227CFC41E0E14162B158F441CE3969F43B4EF341C0DC51070EF1B94124DB1B9667FAF941CC0C863DB938F04110DBC29A3F9FC4C16666DD4D6C73F441A4E3089C50C5F3C1DD551D1F853EFEC18824400D1303F2411C69E5DE4874F941010200008008000000EF9BD9ADB340F6C1761DE18F8511EDC1BF991BCAA4BCFDC12BBD85A16DD8F1C1F45E4B4559B9F0C1BA42E567CE90F7415A68717EE9BAE3C102E1DB32AA12E7C1D0E06B1298F0D9C1507D6E148B6EE4C119708BC138B8F0C138DDA33438A1E9C1EC351A411AF2FB41D65DC24926A0F8C1B0043287990DDA41A0E21816BE94FD4180A98CC0BAAD9B41CA7AF757BAC3004206E471AF2CD8F8419C526EBC149BF7411C0FA0AA4F41F9C1F03656DBD3F0C6C1D47CE1AB4A61E641FCCB9EBC3596FC41010200008004000000568D5007BC8100423053785E2D01CAC184332FCE51F9FC41E4179DD86A84EE41C020CE98A5A2D941E07F65090DF501C2F0043D8C30C6E3C12080C258CC70F0417A7FD4DE1F55F4C16D1C1F076628F4C1D46A03545A99FD41CC6092568515F84101050000800900000001020000800200000097FF7F99C97DF2C185D61B3F8333F1C1E9A50D1C04C8FBC15094E8BA01EBE7411E2420C68574E9C1004C78C91BA8E6C1010200008003000000801204F9A88BF1418C715DAA3790F0C181A8FB8D4E23FCC1CC015A56D23AFB419C03256A4C5EF741DEEC9079AC53F341004A5694E68576C13407ED94A917FB412C5E8F753D25EE410102000080050000007EEC3DA109BFF1C12C07D5037B3BD6C13FF59C697EC501C23BC4B8B60D7B01C26FE9584FAD09FDC15C5F8E26DA95E0C1AE524CDC94F7F0C1E5469EA724C3F2C1017046F3CCE1F0C1A837D84526D5E84166F8CC6BA188F6C1882F97EF9D06F54130DBA9585261F6416FED3DF094DBF1C1209D16935EE3D7C1010200008005000000AC55F30D4BE0FF41203AE96C984EE1C108EA334832BBE141300AD45BC4D7F941007C5E51FACEFA41247EA934CFD7E9412C441C2DDC9CFD41888211040969014258149450DB36E141E0291008687FD24120E5D13032D50042A6C32C08CCBCE7C11CBB5AC4FA48FDC10CAAEFD1906E014274EF9D77D20C0142010200008003000000A8BA3FC34220DEC17853AC8AB230E7C1DA05107FB8EEF6C1BCD0336D4F19F2417836F3D781B5F041A4B30FB50127F641C85C5DD0626CF8C13AD8609190B401425E27AC19EE64F54101020000800300000014D67F03F86EF0418E01FFA87A34F641B0AC3F3575F9FD4120D7FB7FCC3BDF41FECF21B1FEC3F241D817C112DDD9D4C1401DCFE3A811AAC1A83D35E50DB3EA410019E3ACE1FAD2C10102000080070000008399266FE38101C27D6348C2DE0802C24CA9BB77E2ECFB41265B80C39309E4C1552E7F0DD915F2C1507930CE4D9CD14100AA520E590CE5C170370FA6039AD0C11A2EAFF3020AF441A406DBDCA54BE04140DA7F7F86BCCAC1606695C8BBECD2C1587EDF46C6B8E141308FE889ECA1F64128A36331A990E9C178BD90663E94D34142C9DEF51E73F341843DB8BBAF2AFB41E0E2A07D447DE5411C321BECE788004278D8E99A4516F1C1010200008004000000C01DF7C1D6C001C2986C96661AD7DEC11B1B38FDD202F2C1402D077D1BE2C641EC72FB80EA06EAC13F28399B1778F2C174C0B657915AE5412A2D1810C349F8C19067982DDA47CFC1B0562113C9B9E541CCC0629AE374FE41044780C32824FC41010200008004000000C0340C967D66C141BC912F4FE495D9C14C7B4FC2B474E241B075462D8E0EFC41ACE08BCFA8BFE7C10818BD7DB55AE34118B2CE2883E0F8412011910F58B5EF41F0927F68A7E4CDC10CCC8380DB95F2C17C050744FE56EA41CA6BC3641686004201010000807090282D7E24C24128498D726B6601C240163F6FA307E241':::GEOMETRY
								)
						)
							AS tab_79206 (col_180415)
				)
		SELECT
			count(*) AS col_180416
		FROM
			defaultdb.public.table1@[0] AS tab_79207
		WHERE
			overlaps(localtimestamp(tab_79207.col1_4::INT8):::TIMESTAMPTZ::TIMESTAMPTZ::TIMESTAMPTZ, '2004-05-17 12:24:11.000547+00:00':::TIMESTAMPTZ::TIMESTAMPTZ, '1970-01-20 01:53:09.000019+00:00':::TIMESTAMPTZ::TIMESTAMPTZ, '2029-05-09 02:21:44.000306+00:00':::TIMESTAMPTZ::TIMESTAMPTZ)::BOOL
		GROUP BY
			tab_79207.col1_1
		HAVING
			every(true::BOOL)::BOOL
		LIMIT
			34:::INT8;
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=rand-tables/setting=no-mutations.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/cmd/roachtest/tests/sqlsmith.go
+++ b/pkg/cmd/roachtest/tests/sqlsmith.go
@@ -252,12 +252,10 @@ INSERT INTO seed_mr_table DEFAULT VALUES;`, regionList[0]),
 				es := err.Error()
 				if strings.Contains(es, "internal error") {
 					// TODO(yuzefovich): we temporarily ignore internal errors
-					// that are because of #40929 and #86009.
+					// that are because of #40929.
 					var expectedError bool
 					for _, exp := range []string{
 						"could not parse \"0E-2019\" as type decimal",
-						"unable to vectorize execution plan: localtimestamp",
-						"unable to vectorize execution plan: overlaps",
 					} {
 						expectedError = expectedError || strings.Contains(es, exp)
 					}

--- a/pkg/sql/logictest/testdata/logic_test/distsql_builtin
+++ b/pkg/sql/logictest/testdata/logic_test/distsql_builtin
@@ -1,0 +1,16 @@
+# LogicTest: 5node-default-configs
+
+statement ok
+CREATE TABLE t (c INT);
+
+statement ok
+ALTER TABLE t EXPERIMENTAL_RELOCATE VALUES (ARRAY[2], 2)
+
+# Regression test for not propagating the type annotation on the builtin
+# function with multiple overloads when serializing the expression for DistSQL
+# (#86009).
+statement ok
+SELECT c FROM t
+WHERE
+	localtimestamp(7::INT8):::TIMESTAMPTZ
+	IN ('1975-04-24 08:08:35.000071+00:00':::TIMESTAMPTZ, '1980-10-15 12:17:59.000616+00:00':::TIMESTAMPTZ);

--- a/pkg/sql/logictest/tests/5node-disk/BUILD.bazel
+++ b/pkg/sql/logictest/tests/5node-disk/BUILD.bazel
@@ -10,7 +10,7 @@ go_test(
         "//c-deps:libgeos",  # keep
         "//pkg/sql/logictest:testdata",  # keep
     ],
-    shard_count = 13,
+    shard_count = 14,
     tags = ["cpu:3"],
     deps = [
         "//pkg/build/bazel",

--- a/pkg/sql/logictest/tests/5node-disk/generated_test.go
+++ b/pkg/sql/logictest/tests/5node-disk/generated_test.go
@@ -93,6 +93,13 @@ func TestLogic_distsql_agg(
 	runLogicTest(t, "distsql_agg")
 }
 
+func TestLogic_distsql_builtin(
+	t *testing.T,
+) {
+	defer leaktest.AfterTest(t)()
+	runLogicTest(t, "distsql_builtin")
+}
+
 func TestLogic_distsql_crdb_internal(
 	t *testing.T,
 ) {

--- a/pkg/sql/logictest/tests/5node/generated_test.go
+++ b/pkg/sql/logictest/tests/5node/generated_test.go
@@ -93,6 +93,13 @@ func TestLogic_distsql_agg(
 	runLogicTest(t, "distsql_agg")
 }
 
+func TestLogic_distsql_builtin(
+	t *testing.T,
+) {
+	defer leaktest.AfterTest(t)()
+	runLogicTest(t, "distsql_builtin")
+}
+
 func TestLogic_distsql_crdb_internal(
 	t *testing.T,
 ) {

--- a/pkg/sql/sem/tree/function_definition.go
+++ b/pkg/sql/sem/tree/function_definition.go
@@ -189,8 +189,11 @@ func NewFunctionDefinition(
 		if def[i].PreferredOverload {
 			// Builtins with a preferred overload are always ambiguous.
 			props.AmbiguousReturnType = true
+			break
 		}
+	}
 
+	for i := range def {
 		def[i].FunctionProperties = *props
 		overloads[i] = &def[i]
 	}

